src/Storages/Streaming/StorageRandom.cpp

```diff
@@ -468,25 +487,32 @@ class GenerateRandomSource final : public ISource
 
         if (!is_streaming)
         {
-            auto batch_size = std::min(block_size, total_events - generated_events);
+            auto batch_size = std::min(max_block_size, total_events - generated_events);
             generated_events += batch_size;
 
-            /// random stream table query will return a block_size of chunk and end query.
+            /// random stream table query will return a max_block_size of chunk and end query.
             return doGenerate(batch_size);
         }
 
-        if (events_per_second != 0)
+        if (events_per_second != 0 || slow_eps)
```

Comment:
if slow_eps is true, the `events_per_second` must be true.  remove it

Reply:
it's complex and unreadable. we should use separately functions to prcess multiple generate modes here.